uppsrc/Core/SSH/Session.cpp

```diff
@@ -205,32 +205,12 @@ bool SshSession::Connect(const String& host, int port, const String& user, const
 			if(!rc) {
 				LLOG("Handshake successful.");
 				WhenPhase(PHASE_AUTHORIZATION);
+				if(WhenVerify && !WhenVerify(host, port))
+					ThrowError(-1);
```

Comment:
ThrowError(-1) is enigmatic. It should ThrowError information...

Reply:
@klugier 

It is a *generic failure* or rejection, separated from the libssh2's own error codes. Since the WhenVerify() hands the operation over to the user. (client). Client *should* deal with it (i.e. it can print a cancellation message or popup a error dialog.